pkgs/servers/monitoring/loki/default.nix

```diff
@@ -54,7 +54,7 @@ buildGoModule rec {
     description = "Like Prometheus, but for logs";
     license = with licenses; [ agpl3Only asl20 ];
     homepage = "https://grafana.com/oss/loki/";
-    maintainers = with maintainers; [ willibutz globin mmahut ];
+    maintainers = with maintainers; [ willibutz globin mmahut indeednotjames ];
     platforms = platforms.unix;
```

Comment:
```suggestion
```
already set by buildGoModule

Reply:
From what I can tell, `lib.platforms.darwin ++ lib.platforms.linux` != `lib.platforms.unix`.
Can you elaborate?

```bash
# platforms.unix
❯ nix eval --json .#grafana-loki.meta.platforms | jq 'length'
41
# unset
❯ nix eval --json .#grafana-loki.meta.platforms | jq 'length'
22
```
https://github.com/NixOS/nixpkgs/blob/b1581c36189e5063163acb24822ed8a343bf6a8f/pkgs/build-support/go/module.nix#L320
https://github.com/NixOS/nixpkgs/blob/b1581c36189e5063163acb24822ed8a343bf6a8f/pkgs/development/compilers/go/1.20.nix#L180

Reply:
Go only support darwin and linux and if that ever changes in the future, we likely do not need to update meta.platforms for every package.

Reply:
Took a decent while to compile from scratch:
```bash
❯ nix-build -A pkgsCross.x86_64-freebsd.grafana-loki
/nix/store/3l3irkk9aw6h9dgx6qn5nmik4bym5g6v-grafana-loki-x86_64-unknown-freebsd13-2.7.4
❯ file result/bin/*
result/bin/logcli:      ELF 64-bit LSB executable, x86-64, version 1 (FreeBSD), dynamically linked, interpreter /libexec/ld-elf.so.1, Go BuildID=6lZq-V9UeCRKTYdXek8f/NYD_WYwJaCF580YhsjNI/9jb1AW3bhPtPnNLcWZuF/PXcUiWiKviscMaEru4wZ, stripped
result/bin/loki:        ELF 64-bit LSB executable, x86-64, version 1 (FreeBSD), dynamically linked, interpreter /libexec/ld-elf.so.1, Go BuildID=tRS_s9fFV4EnYqCD83qt/ZXynWpT4OP_AN_Z5LsWK/Wsy56FglfdPorzZlKkBb/vEJ8cl8xa-LW40dAYBOK, stripped
result/bin/loki-canary: ELF 64-bit LSB executable, x86-64, version 1 (FreeBSD), dynamically linked, interpreter /libexec/ld-elf.so.1, Go BuildID=Bc4Woh_7Wyc4TM3sFUqW/QAat7p3fnb58-WEv_5i6/xq-ITXlM1oRq4KIh1H_6/NeMsLmDRD8THPpAq1USx, stripped
result/bin/promtail:    ELF 64-bit LSB executable, x86-64, version 1 (FreeBSD), dynamically linked, interpreter /libexec/ld-elf.so.1, Go BuildID=v8sMpP1An8VjLW799gya/uM8d1x6OZOSnXdSD30Eb/ApOIDl0t5Ux9gsggToa6/X14UW_THl-9d642WG0g3, stripped
```

Upstream officially supports freebsd as target, and I see no reason why nixpkgs should drop it.

Is anyone actually using nixpkgs to compile loki for freebsd?
Lol no, extremely unlikely.

Should we actively prevent it?
No, I don't think so.

Reply:
> Upstream officially supports freebsd as target, and I see no reason why nixpkgs should drop it.

Our go does not support freebsd. If that supports freebsd it will be automatically added to here

Reply:
I can't quite follow you, but whatever.

Then why don't you do a treewide change?
With this rather primitive query there are seemingly 141 files using `buildGoModule` and `platforms.unix`.
https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+buildGoModule+%22platforms+%3D%22+unix&type=code